pkg/pgmodel/ingestor/ingestor.go

```diff
@@ -24,19 +24,24 @@ type Cfg struct {
 
 // DBIngestor ingest the TimeSeries data into Timescale database.
 type DBIngestor struct {
-	dispatcher model.Dispatcher
-	sCache     cache.SeriesCache
+	sCache             cache.SeriesCache
+	samplesDispatcher  model.Dispatcher
+	metadataDispatcher model.Dispatcher
```

Comment:
I don't think metadata needs to conform to the `model.Dispatcher` interface. There is no clear reason to do it since the data is not comparable. Changing the interface to accept an empty interface removes the type safety without any gain.

Metadata needs a simple inserter (just like the one you made here) without implementing the dispatcher interface. I would prefer it if you changed `model.Dispatcher` by adding another method `InsertMetadata`. 

Then it would be simple to keep everything under `pgxDispatcher` and makes a lot more sense.